### PR TITLE
Omissions in types supported by both Desktop and Core - System.Security.AccessControl.AuthorizationRuleCollection

### PR DESCRIPTION
**Pull request by ViktorHofer**
Adjustments for System.Security.AccessControl.AuthorizationRuleCollection to correctly inherit from System.Collections.NonGeneric.ReadOnlyCollectionBase. For more info see here: https://github.com/dotnet/corefx/issues/15255#issuecomment-289550875

### CODE REVIEW

File: src/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj [hunk 1]
```diff
@@ -34,6 +34,7 @@
     <Compile Include="System\IO\FileSystemAclExtensions.net46.cs" />
   </ItemGroup>
   <ItemGroup Condition="'$(TargetGroup)'!='netfx'">
+	<Reference Include="System.Collections.NonGeneric" />
```
Comment (danmosemsft) [inline src/System.IO.FileSystem.AccessControl/src/System.IO.FileSystem.AccessControl.csproj]: Nit: Indentation

File: src/System.Security.AccessControl/src/System.Security.AccessControl.csproj [hunk 2]
```diff
@@ -120,6 +120,7 @@
   </ItemGroup>
   <ItemGroup>
     <Reference Include="System.Collections" />
+    <Reference Include="System.Collections.NonGeneric" />
```
Comment (danmosemsft) [inline src/System.Security.AccessControl/src/System.Security.AccessControl.csproj]: Before we merge this -- @joperezr @weshaggard 

S.S.AccessControl builds for `netstandard` but S.C.NonGeneric builds for `netcoreapp` and `uap` specifically. So assume this reference is prohibited. But why does NonGenreic not build for `netstandard` only?

Comment (danmosemsft) [inline src/System.Security.AccessControl/src/System.Security.AccessControl.csproj]: Oh, it's because it depends on `System.Security.Principal.Windows`.

Comment (danmosemsft) [inline src/System.Security.AccessControl/src/System.Security.AccessControl.csproj]: I'm missing something conceptually -- I guess types in netstandard can be defined in libraries that don't have a netstandard configuration. That makes sense I guess. So is this new reference legal?

Comment (joperezr) [inline src/System.Security.AccessControl/src/System.Security.AccessControl.csproj]: S.S.AccessControl doesn't build for netstandard, at least not the implementation. The implementation is also platform-specific (uap, netcoreapp) which is why this is a valid reference.

Comment (danmosemsft) [inline src/System.Security.AccessControl/src/System.Security.AccessControl.csproj]: That's what confused me -- the ref is built for netstandard. Why's that? Do we just do that for convenience whenever it happens to compile successfully for netstandard?

Comment (weshaggard) [inline src/System.Security.AccessControl/src/System.Security.AccessControl.csproj]: We should actually consider switching this library to be netstandard but if we do that then we may need to set some explicit configurations to Generate platform not supported exceptions (i.e. unix and maybe uap).

Comment (ViktorHofer) [inline src/System.Security.AccessControl/src/System.Security.AccessControl.csproj]: should we merge the PR for now and track these ideas in a separate issue?

**Comment by weshaggard** [comment]
@ViktorHofer it looks like you didn't correctly merge with the latest from master as your PR shows a lot of files which aren't relevant to your changes.

**Comment by ViktorHofer** [comment]
@weshaggard yes, alread on it...

**Comment by ViktorHofer** [comment]
@weshaggard Tests should finish soon. Please take another look. As said above if you agree I would finish the PR as it is now and open a new issue with what you suggested.

File: src/System.Security.AccessControl/src/System.Security.AccessControl.csproj (hunk 2 above)
Comment (weshaggard) [inline src/System.Security.AccessControl/src/System.Security.AccessControl.csproj]: I'm fine with merging this once CI is green. We do need to file an issue about all the AccessControl APIs about whether or not we want to support then for netstandard or not. @joperezr was going to file a similar issue, so chat with him about it.

Comment (joperezr) [inline src/System.Security.AccessControl/src/System.Security.AccessControl.csproj]: I logged: #17660

**Pull request merged**